src/routes/courses/handlers.ts

```diff
@@ -28,6 +32,84 @@ export async function allCourses(req: Request, res: Response) {
   }
 }
 
+export async function getCourseById(req: Request, res: Response) {
+  try {
+    const { id } = req.params;
+    const course = await getById(Number(id));
+    res.json(course);
+  } catch (err) {
+    await handleErrors(err, req, res, logContext);
+  }
+}
+
+export async function updateCourseById(req: Request, res: Response) {
+  try {
+    const { id } = req.params;
+    const userId = await currentUserId(req, res);
+    const user = await userById(userId);
+    const authorization = new UserPolicy(user);
+
+    if (!authorization.isAdmin()) {
+      res.status(403).send('Forbidden');
+      return;
+    }
+
+    const course = await getById(Number(id));
+    if (!course) {
+      res.status(404).send('Course not found');
+      return;
+    }
+    const updated = await course.update(req.body);
+    res.json(updated);
+  } catch (err) {
+    await handleErrors(err, req, res, logContext);
+  }
+}
+
+export async function createCourseByName(req: Request, res: Response) {
+  try {
+    const userId = await currentUserId(req, res);
+    const user = await userById(userId);
+    const authorization = new UserPolicy(user);
+
+    if (!authorization.isAdmin()) {
+      res.status(403).send('Forbidden');
+      return;
+    }
+
+    const { name } = req.body;
+    const course = await createCourse(name);
+
+    res.json(course);
+  } catch (err) {
+    await handleErrors(err, req, res, logContext);
+  }
+}
+
+export async function deleteCourseById(req: Request, res: Response) {
+  try {
+    const { id } = req.params;
+    const userId = await currentUserId(req, res);
+    const user = await userById(userId);
+    const authorization = new UserPolicy(user);
+
+    if (!authorization.isAdmin()) {
+      res.status(403).send('Forbidden');
+      return;
+    }
+
+    const course = await getById(Number(id));
+    if (!course) {
+      res.status(404).send('Course not found');
+      return;
+    }
+    await course.destroy();
```

Comment:
this will fail if the course is already used, and there for linked to a report, right?

Reply:
Yeah, good point. Catching and responding appropriately now: 2539aa5e83ca7ed4207a11c34127c7af954ad8b9